### DISCUSSION TITLE
Imagick PHP - Getting DPI for AI file

### DISCUSSION THREAD

**Discussion by rayzistdev** [ImageMagick/ImageMagick#2684] [Help]
I want to get the resolution of an AI file but it only returns 72. However, the resolution is supposed to be 300. If I were to change the resolution of the image using imagick, it would come out to be something far less [the dimensions]. How can I get the actual resolution of a file?

In case I can't use imagick for that, is there a library or framework that can give me the precised resolution of a file (regardless what type)?

I have my code as:

```
try {
        $imagick = new Imagick($imagePath);
        $data = $imagick->identifyimage();
    }
    catch(Exception $e) {
        echo json_encode('ERROR: ' . $e->getMessage()); 
    }
    
    if($data['units'] == 'PixelsPerCentimeter') {
        $imageResolution = $imagick->getImageResolution(); // Temp Image Resolution
        //echo 'pixels per cent lol';
        if (!empty($imageResolution['y'])) {
            $dpi = intval( round($imageResolution['y'] * 2.54, 2) );
        }
      }
    else $dpi = $data['resolution']['x'];
  
    $width = round($data['geometry']['width'] / $dpi, 3);
    $height = round($data['geometry']['height'] / $dpi, 3);
```

**Comment by fmw42** [comment]
AI file is a vector file. It has no resolution. It reads 72, as a default for use when converting to a raster file. You can make the resolution anything you want, when you read the AI file and convert it to some raster file format.

**Comment by snibgo** [comment]
As Fred says. How about asking exiftool? This always seems to return:
```
AI Target Resolution            : 800
```
